### PR TITLE
allow user to explicitly track values for "exists" query

### PR DESCRIPTION
**Pull request by gnusi**
Please note that for legal reasons we require you to sign the 
[Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/401/

File: arangod/IResearch/IResearchView.cpp [hunk 1]
```diff
@@ -275,7 +275,12 @@ inline void insertDocument(
 
   // User fields
   while (body.valid()) {
-    doc.insert(irs::action::index_store, field);
+    if (ValueStorage::NONE == field._storeValues) {
+      doc.insert(irs::action::index, field);
+    } else {
```
Comment (jsteemann) [inline arangod/IResearch/IResearchView.cpp]: the `else` case covers types `ID` and `FULL`, but is identical for both. Is it intentional?

Comment (gnusi) [reply arangod/IResearch/IResearchView.cpp]: `FULL` is not supported at the moment and meant to be treated as the reserved value for future usage

Comment (jsteemann) [reply arangod/IResearch/IResearchView.cpp]: ah, ok!

**Pull request merged**